pkg/corerp/api/v20231001preview/application_conversion.go

```diff
@@ -26,6 +26,15 @@ import (
 // ConvertTo converts from the versioned Application resource to version-agnostic datamodel.
 func (src *ApplicationResource) ConvertTo() (v1.DataModelInterface, error) {
 	// Note: SystemData conversion isn't required since this property comes ARM and datastore.
+
+	if src.Properties == nil {
```

Comment:
If we update the typespec file to mark `properties` as required, do we still need to do this?

Reply:
No, this check wont be needed  then.

Reply:
I could not find a way to make the entire properties block as required. Found docs only on the individual properties using "?". I have sent a note to Mike Kistler on the typespec team who had joined one of our review meetings.

Reply:
Response from typespec team: The optionality and visibility of the RP-specific 'properties' bag is standardized for all RPs, this ensure that they all function similarly.  In this case, the top level properties are intentionally left optional for everyone - this does not mean that the properties bag is not required for creating the resource, or that it can ever be replace with 'null',  but there are certain PUT requests, such as 'envelope property updates, particularly for Tags that require the properties bag to be specified as optional.

Reply:
That's not a very helpful response. Isn't this also impacting the bicep authoring experience?

Reply:
**Example: Before Completion**

<img width="1069" alt="image" src="https://github.com/radius-project/radius/assets/1430011/978fd4fb-dea6-452b-b8fe-88499ad72fbd">

**Example: After Completion**

<img width="661" alt="image" src="https://github.com/radius-project/radius/assets/1430011/d393475d-d7a4-4fd4-a462-78e1f80bf0d1">

Reply:
This used to complete the following before we migrated to TypeSpec:

```bicep
resource myrecipe 'Applications.Datastores/redisCaches@2023-10-01-preview' = {
  name: 
  properties: {
    environment: 
    application:
  }
}
```

Reply:
@rynowak I agree that this should be handled by typespec.. for now, do we want to move forward with this validation on the backend, and follow up further with the typespec team on their reasoning? @vinayada1

Reply:
> I would say let's follow up with the typespec team separately. That way we are at least protected against this panic in our code in some way immediately.

Presumably **every** resource in Radius now has this bug thanks to the Typespec change. If we're going to make a fix it's a code change in **every** resource. 

This is why I don't think it's reasonable.